utils/build-script-impl

```diff
@@ -101,8 +97,6 @@ KNOWN_SETTINGS=(
     installable-package         ""               "the path to the archive of the installation directory"
     test-installable-package    ""               "whether to run post-packaging tests on the produced package"
     reconfigure                 ""               "force a CMake configuration run even if CMakeCache.txt already exists"
-    build-libparser-only        ""               "only build libSwiftSyntaxParser"
```

Comment:
Why are we removing `--build-libparser-only`?  Isn't is still useful to be able to build the parser library without building the rest of the compiler?  CC @akyrtzi

Reply:
This has been superseded by running `utils/build-parser-lib`